yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -98,6 +121,9 @@ class YarnClusterSuite extends FunSuite with BeforeAndAfterAll with Matchers wit
     }
 
     fakeSparkJar = File.createTempFile("sparkJar", null, tempDir)
+    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))
```

Comment:
Because PythonRunner should use sparkPythonPath that contains pyspark library, here i  add SPARK_HOME to env of executor and driver. if we do not add this env, python in executor or driver cannot work. @sryza